src/ray/core_worker/core_worker.cc

```diff
@@ -2568,6 +2582,43 @@ void CoreWorker::HandleDirectActorCallArgWaitComplete(
   send_reply_callback(Status::OK(), nullptr, nullptr);
 }
 
+void CoreWorker::HandleGetObjectWorkingSet(const rpc::GetObjectWorkingSetRequest &request,
+										   rpc::GetObjectWorkingSetReply *reply,
+                                       rpc::SendReplyCallback send_reply_callback) {
+	RAY_LOG(DEBUG) << "[JAE_DEBUG] HandleGetObjectWorkingSet called" ;
+	std::vector<ObjectID> obj_ids;
+	std::vector<ObjectID> deleted_obj_ids;
+	for(int i=0; i<request.object_ids_size(); i++){
+	  obj_ids.push_back(ObjectID::FromBinary(request.object_ids(i)));
+	  RAY_LOG(DEBUG) << "[JAE_DEBUG] HandleGetObjectWorkingSet obj in obj_store:" << ObjectID::FromBinary(request.object_ids(i));
+	}
+	for(int i=0; i<request.deleted_object_ids_size(); i++){
+	  ObjectID obj = ObjectID::FromBinary(request.deleted_object_ids(i));
+	  RAY_LOG(DEBUG) << "[JAE_DEBUG] HandleGetObjectWorkingSet deleted objects:" << obj;
+	  deleted_obj_ids.push_back(obj);
+	  object_working_set_.erase(obj);
+	  for(auto &working_set : object_working_set_){
+		working_set.second.erase(obj);
+	  }
+	}
+
+	//std::vector<ObjectID> gcable_objs;
+	for(auto &working_set : object_working_set_){
+		  RAY_LOG(DEBUG) << "[JAE_DEBUG] objects_working Set obj " << working_set.first;
+		for(auto &itr: working_set.second){
+		  RAY_LOG(DEBUG) << " :" << itr;
+		}
+		if(std::includes(obj_ids.begin(), obj_ids.end(),
```

Comment:
Can we add a comment and unit test for this?

Reply:
It will be easiest to unit test this if you move it to a separate class. Also needs a lock.